### PR TITLE
Attachement live preview

### PR DESCRIPTION
**Pull request by rabuzarus**
addresses https://github.com/friendica/friendica/issues/6535

This PR implements a jot live preview for attachments. Therefor I rewrote much of LeonardoCardoso plugin https://github.com/LeonardoCardoso/Facebook-Link-Preview/blob/master/js/linkPreview.js

At the moment seems to work very well.
There would be some things for the future:
- at the moment it only implemented for frio (it very easy to make it work with the other theme but it would need time to style it with css)
- at the moment only `[attachment]` is handled - images, audio, videos (non attachment) are inserted into the jot as bbcode - we need a discussions about what we want to preview and how much.
- `/mod/parse_url.php` seems to return attachment data even when the page isn't available - @annando can we change this?
- ~~for frio I would suggest to have a max with of the jot modal and to make the content inside the modal scroll-able (like it is done at g+ modal) - because the jot formatting buttons are at the bottom of the jot modal (something in css with `calc()` and `vh` as unit - if someone wants to try, feel free to do it)~~
- if the attachment type is video the live preview divers from the look of the real item. This is because we don't get the oembed thumbnail dimensions through `parse_url`.  If we want to enhance the behavior we need to discuss how we want to make the oembed data accessable
- if edition an item frio can't distinguish if it is an top-level-post or an comment - that's why the attachment preview does also work if a comment is edited. But there shouldn't be any attachments in comments

At the moment I wont be able to work on this (this feature - and especially the polishing - took already longer than I thought). But I think that we have with this jQuery plugin a platform, where anybody can add features.

Here a video from the live preview in action
https://owncloud.kommune4.de/index.php/s/dJDkZpE6RoLtkdx

### CODE REVIEW

File: view/theme/frio/js/jot.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// We apptend the linkPreview to a global Variable to make linkPreview
+// accessable on other places. Note: search on other places before you
+// delete or move the variable.
+var linkPreview;
```
[the remark this answers is not in the document]
Comment (rabuzarus) [reply view/theme/frio/js/jot.js]: we need this variable to call the public functions `crawlText()` `addBBCodeToPreview()`, and `destroy()`
e.g. in jot.js `linkPreview.crawlText(reply + noAttachment);''

[the remark this answers is not in the document]
Comment (MrPetovan) [reply view/theme/frio/js/jot.js]: That's my point, can't we use `$('#profile-jot-text').linkPreview()` in these calls?

[the remark this answers is not in the document]
Comment (MrPetovan) [reply view/theme/frio/js/jot.js]: I'm pretty sure that you can call `$('#profile-jot-text').linkPreview()` repeatedly and only initialize the plugin once. At least this is the jQuery plugin philosophy.

[the remark this answers is not in the document]
Comment (rabuzarus) [reply view/theme/frio/js/jot.js]: I have tested it - everytime you run `$('#profile-jot-text').linkPreview()` the plugins `init()` function is called and the variables are set to the default values

[the remark this answers is not in the document]
Comment (MrPetovan) [reply view/theme/frio/js/jot.js]: Then there's something wrong with the plugin. I appreciate that you found a workaround to make it work anyway but ultimately this is the wrong approach.

[the remark this answers is not in the document]
Comment (rabuzarus) [reply view/theme/frio/js/jot.js]: have a look into the PR description - there is a link to the original source

[the remark this answers is not in the document]
Comment (MrPetovan) [reply view/theme/frio/js/jot.js]: Ha, thanks, I only looked in the JS file.

**Comment by annando** [comment]
@rabuzarus I hadn't got any time at all to look into the code. I guess you are called the "parse_url" endpoint via JS? I think we could/should add an option to return an empty value upon failure.

**Comment by annando** [comment]
What is left to do, before we can accept this PR?

**Comment by MrPetovan** [comment]
I wanted to get rid of the global variable in the Javascript file but it looks like the plug-in itself is wrongly structured. This would require significantly more work to make it compliant with jQuery design standards.

**Pull request merged**